File: app/src/main/java/wmding/example/com/mylivetools/module/search/SearchContract.java
package wmding.example.com.mylivetools.module.search;

import android.support.annotation.NonNull;

import java.util.List;

import wmding.example.com.mylivetools.base.BasePresenter;
import wmding.example.com.mylivetools.base.BaseView;
import wmding.example.com.mylivetools.bean.ArticleDetailData;

/**
 * Created by wmding on 2018/9/5.
 */
public interface SearchContract {

    interface Presenter extends BasePresenter {


        /**
         * 搜索文章
         * @param page
         * @param keyWords
         * @param forceUpdate
         * @param clearCache
         */
        void searchArticles(@NonNull int page, @NonNull String keyWords, boolean forceUpdate, boolean clearCache);

    }

    interface View extends BaseView<Presenter> {

        /**
         * 隐藏键盘
         */
        void hideKeyboard();

        /**
         * 页面是否是活动的
         *
         * @return
         */
        boolean isActive();


        /**
         * 显示文章列表
         *
         * @param articlesList
         */
        void showArticles(List<ArticleDetailData> articlesList);

        /**
         * 是否显示空白页面
         *
         * @param isShow
         */
        void showEmptyView(boolean isShow);


    }


}

File: app/src/main/java/wmding/example/com/mylivetools/module/search/SearchContract.kt
package wmding.example.com.mylivetools.module.search

import wmding.example.com.mylivetools.base.BasePresenter
import wmding.example.com.mylivetools.base.BaseView
import wmding.example.com.mylivetools.bean.ArticleDetailData

/**
 * Created by wmding on 2018/9/5.
 */
interface SearchContract {

    interface Presenter : BasePresenter {


        /**
         * 搜索文章
         * @param page
         * @param keyWords
         * @param forceUpdate
         * @param clearCache
         */
        fun searchArticles(page: Int, keyWords: String, forceUpdate: Boolean, clearCache: Boolean)

    }

    interface View : BaseView<Presenter> {

        /**
         * 页面是否是活动的
         *
         * @return
         */
        val isActive: Boolean

        /**
         * 隐藏键盘
         */
        fun hideKeyboard()


        /**
         * 显示文章列表
         *
         * @param articlesList
         */
        fun showArticles(articlesList: List<ArticleDetailData>)

        /**
         * 是否显示空白页面
         *
         * @param isShow
         */
        fun showEmptyView(isShow: Boolean)


    }


}
